x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/crawl_details_flyout/accordion_list.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import {
+  EuiAccordion,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiInMemoryTable,
+  EuiNotificationBadge,
+  EuiTitle,
+  IconType,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+
+import './accordion_list.scss';
+
+interface Props {
+  iconType: IconType;
+  items: string[];
+  hasBorder?: boolean;
+  initialIsOpen?: boolean;
+  title: string;
+}
+
+export const AccordionList: React.FC<Props> = ({
+  iconType,
+  items,
+  hasBorder,
+  initialIsOpen,
+  title,
+}) => {
+  const accordionId = useGeneratedHtmlId({
+    prefix: 'accordionList',
+  });
+
+  return (
+    <EuiAccordion
+      initialIsOpen={initialIsOpen}
+      arrowProps={{
+        isDisabled: items.length === 0,
+      }}
+      className={hasBorder ? 'appSearchAccordion--bordered' : 'appSearchAccordion'}
+      buttonContent={
+        <EuiFlexGroup direction="row" responsive={false} gutterSize="s" alignItems="center">
+          <EuiFlexItem grow={false}>
+            <EuiIcon type={iconType} />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiTitle size="s">
+              <h3>{title}</h3>
+            </EuiTitle>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      }
+      id={accordionId}
+      extraAction={<EuiNotificationBadge size="m">{items.length}</EuiNotificationBadge>}
+    >
+      <EuiInMemoryTable
+        items={items.map((item) => ({ item }))}
+        columns={[
+          {
+            render: ({ item }: { item: string }) => item,
+          },
+        ]}
+        pagination={
+          items.length > 10
```

Comment:
I think the value `10` here should be extracted to a constant or perhaps even better with a prop (with a default value of 10).

Reply:
Handled with an optional `rowCount` prop set to a default of `10` 👍🏼